docs/source/futures.rst

```diff
@@ -267,6 +267,15 @@ Or collect all futures in batches that had arrived since the last iteration:
       for future, result in batch:
           ...
 
+In all of the above mentioned cases, it may be advisable to also remove the
+reference to the future in the futures list, otherwise the scheduler will
+keep the results in memory.
+
+.. code-block:: python
+    for future in as_completed(futures):
+        result = future.result()
+        futures.remove(future)
```

Comment:
This is useful in some cases, but not in others.  My sense is that this documentation snippet would solve your problem, but might confuse others.  The amount that novice users blindly copy-paste and get themselves in trouble is surprising :)

Instead, I wonder if we might link to some general documentation about memory management.